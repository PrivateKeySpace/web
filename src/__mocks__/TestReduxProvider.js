import React from 'react'
import PropTypes from 'prop-types'
import { createStore } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'
import rootReducer from '../store/rootReducer'

const displayName = '__mocks__/TestReduxProvider'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ state: PropTypes.object, children: PropTypes.any }),
  defaultProps({ state: {} })
)

function TestReduxProvider (props) {
  const { state, children } = props
  const store = createStore(rootReducer, state)

  return (
    <ReduxProvider store={store}>
      {children}
    </ReduxProvider>
  )
}

export default enhance(TestReduxProvider)
