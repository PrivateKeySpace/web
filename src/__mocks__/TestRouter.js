import React from 'react'
import PropTypes from 'prop-types'
import { MemoryRouter } from 'react-router'
import { compose, setDisplayName, setPropTypes, defaultProps } from 'recompose'

const displayName = '__mocks__/TestRouter'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes({ path: PropTypes.string, children: PropTypes.any }),
  defaultProps({ path: '/' })
)

function TestRouter (props) {
  const { path, children } = props

  return (
    <MemoryRouter initialEntries={[path]} initialIndex={0}>
      {children}
    </MemoryRouter>
  )
}

export default enhance(TestRouter)
