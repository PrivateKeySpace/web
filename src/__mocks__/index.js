export { default as TestRouter } from './TestRouter'
export { default as TestReduxProvider } from './TestReduxProvider'
