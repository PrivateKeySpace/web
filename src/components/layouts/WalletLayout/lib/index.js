export { default as propTypes } from './propTypes'
export { default as defaultPropsSpec } from './defaultPropsSpec'
export { default as mapProps } from './mapProps'
