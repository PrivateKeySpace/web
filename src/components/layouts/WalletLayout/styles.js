import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../theme/constants'

const styles = StyleSheet.create({
  container: {
    height: '100vh',
    backgroundColor: colors.white
  },
  topBarContainer: {
    height: 30,
    backgroundColor: colors.danger,
    color: colors.white,
    fontSize: 13,
    textAlign: 'center',
    lineHeight: '30px'
  },
  navContainer: {
    width: 220,
    height: '100%'
  },
  mainContainer: {
    backgroundColor: '#f3f3f3'
  }
})

export default styles
