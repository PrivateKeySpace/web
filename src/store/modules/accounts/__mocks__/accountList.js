const accountList = [
  {
    'path': [
      2147483697,
      2147483648,
      2147483648
    ],
    'address': '3Pc8vh9xUopMmiBW1yDAm3h1aqqbmmv11p',
    'addressPath': [
      2147483697,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6Ca3FEzN1SemuKNujpAeyDLXk7DXXtgpUs5HHLrG5brjVQm7ib3SLqMKfWWQZset7HMm7wM58wVGEJm5eG6rkxEcYvtvExYaM9T7jxkmmMp',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': true
  },
  {
    'path': [
      2147483692,
      2147483648,
      2147483648
    ],
    'address': '1Bswvjfoa1p1kMkwE4C8KAPbYLSKfg1Nni',
    'addressPath': [
      2147483692,
      2147483648,
      2147483648,
      0,
      0
    ],
    'addressId': 0,
    'xpub': 'xpub6D4zdJZuWGKQK5qMyj7reDcapRE6StaVTGS3xWCAjRyLfne84tGxUK1yYNrUWFY8gYtJA7xd3s97o1uiPkFEhpdsuDsNkLnj5iHyAbCEnRv',
    'balance': 0,
    'confirmed': 0,
    'id': 0,
    'segwit': false
  }
]

export default accountList
