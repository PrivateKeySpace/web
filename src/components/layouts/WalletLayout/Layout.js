import React from 'react'
import { compose, setDisplayName, setPropTypes, defaultProps, mapProps, pure } from 'recompose'
import { css } from 'aphrodite/no-important'
import { defaultPropsSpec, propTypes, mapProps as propsMapper } from './lib'
import tx from '../../../theme/styles'
import { Nav, Header } from './components'
import sx from './styles'

const displayName = 'components/layouts/WalletLayout'

const enhance = compose(
  setDisplayName(displayName),
  setPropTypes(propTypes),
  defaultProps(defaultPropsSpec),
  mapProps(propsMapper),
  pure
)

function Layout (props) {
  const {
    topBarContent,
    headerContent,
    mainContent,
    mainContentContainerProps
  } = props
  return (
    <div className={css(tx.flex, tx.flexDirectionColumn, sx.container)}>
      <div className={css(tx.fontWeight500, sx.topBarContainer)}>
        {topBarContent}
      </div>
      <div className={css(tx.flex, tx.flexGrow1, tx.flexDirectionRow, tx.width100)}>
        <div className={css(sx.navContainer)}>
          <Nav />
        </div>
        <div className={css(tx.flexGrow1, sx.mainContainer)}>
          <Header
            headerContent={headerContent}
          />
          <div {...mainContentContainerProps}>
            {mainContent}
          </div>
        </div>
      </div>
    </div>
  )
}

export default enhance(Layout)
