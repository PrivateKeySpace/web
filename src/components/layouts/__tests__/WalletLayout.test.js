import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { TestRouter } from '../../../__mocks__'
import WalletLayout from '../WalletLayout'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const wrapper = render(
    <TestRouter path='/wallet'>
      <WalletLayout
        topBarContent={
          'Test top bar'
        }
        headerContent={
          <h1>
          Test header
          </h1>
        }
        mainContent={
          <div>
            Test main content
          </div>
        }
      />
    </TestRouter>
  )
  expect(wrapper).toMatchSnapshot()
})

it('renders correctly with no top bar defined, does not crash and matches snapshot', () => {
  const wrapper = render(
    <TestRouter path='/wallet'>
      <WalletLayout
        headerContent={
          <h1>
          Test header
          </h1>
        }
        mainContent={
          <div>
            Test main content
          </div>
        }
      />
    </TestRouter>
  )
  expect(wrapper).toMatchSnapshot()
})
