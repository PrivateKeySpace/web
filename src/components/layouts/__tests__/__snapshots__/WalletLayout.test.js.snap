// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no top bar defined, does not crash and matches snapshot 1`] = `
<div
  class="flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-container_ou6sx1"
>
  <div
    class="fontWeight500_gt7myn-o_O-textTransformUppercase_4efw5a-o_O-topBarContainer_1miketn"
  >
    
  </div>
  <div
    class="flex_jro6t0-o_O-flexGrow1_t0tx82-o_O-flexDirectionRow_qb00m9-o_O-width100_zdxht7"
  >
    <div
      class="navContainer_ng807m"
    >
      <nav>
        <a
          href="/wallet/dashboard"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Dashboard
          </div>
        </a>
        <a
          href="/wallet/personal"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Personal accounts
          </div>
        </a>
        <a
          href="/wallet/multisig"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Multisig accounts
          </div>
        </a>
        <a
          href="/wallet/cosigners"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Cosigners
          </div>
        </a>
        <a
          href="/wallet/clients"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Clients
          </div>
        </a>
        <a
          href="/wallet/help"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Help
          </div>
        </a>
      </nav>
    </div>
    <div
      class="flex_jro6t0-o_O-flexGrow1_t0tx82-o_O-flexDirectionColumn_1jtvojh-o_O-mainContainer_1xmqyfw"
    >
      <div
        class="p5_1x929gn-o_O-pt3_1gv8f73-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-width100_zdxht7-o_O-container_1hoplq"
      >
        <div
          class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-width100_zdxht7-o_O-navbar_1u75k1v"
        >
          <h2
            class="textTransformUppercase_4efw5a"
          >
            PKS Wallet
          </h2>
          <div
            class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-height100_1yfus1e-o_O-pullRight_ls0e43"
          >
            <div
              class="mr4_kh8dwt-o_O-fontSize1_3kvou9-o_O-hidden_1u9fru1-o_O-notifications_e296pg"
            >
              <span
                class="lnr lnr-alarm"
              >
                
              </span>
              <div
                class="notificationCounterWrap_1k9zd85"
              >
                <div
                  class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-notificationCounter_1xicdc3"
                >
                  5
                </div>
              </div>
            </div>
            <div
              class="hidden_1u9fru1-o_O-separator_xrxmxy"
            >
              
            </div>
            <div
              class="ml4_1pgo4wm-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-alignItemsFlexEnd_1f2mvon"
            >
              <div
                class="textTransformUppercase_4efw5a-o_O-deviceName_1pe4kle"
              >
                Trezor Label
              </div>
              <div
                class="textTransformUppercase_4efw5a-o_O-deviceStatus_1wddarm"
              >
                connected
              </div>
            </div>
            <div
              class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-px2_6ebohg-o_O-ml1_1dn7v29-o_O-height100_1yfus1e-o_O-cursorPointer_13xlah4-o_O-dropdownButton_1w6iw80"
            >
              <span
                class="lnr lnr-chevron-down icon_ko4whp"
              >
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="pb3_1a70m23-o_O-content_14ddvtv"
        >
          <h1>
            Test header
          </h1>
        </div>
      </div>
      <div
        class="flexGrow1_t0tx82-o_O-flexShrink0_fz3zdn-o_O-p5_1x929gn-o_O-mainContainer_1xmqyfw"
      >
        <div>
          Test main content
        </div>
      </div>
      <footer
        class="px5_1g0n6pg-o_O-footerContainer_14ddvtv"
      >
        <div
          class="width100_zdxht7-o_O-py4_r2hutn-o_O-px1_17oj4c4-o_O-textTransformUppercase_4efw5a-o_O-footer_r1wilx"
        >
          2018 Â© PKS Wallet. All rights reserved.
        </div>
      </footer>
    </div>
  </div>
</div>
`;

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-container_ou6sx1"
>
  <div
    class="fontWeight500_gt7myn-o_O-textTransformUppercase_4efw5a-o_O-topBarContainer_1miketn"
  >
    Test top bar
  </div>
  <div
    class="flex_jro6t0-o_O-flexGrow1_t0tx82-o_O-flexDirectionRow_qb00m9-o_O-width100_zdxht7"
  >
    <div
      class="navContainer_ng807m"
    >
      <nav>
        <a
          href="/wallet/dashboard"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Dashboard
          </div>
        </a>
        <a
          href="/wallet/personal"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Personal accounts
          </div>
        </a>
        <a
          href="/wallet/multisig"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Multisig accounts
          </div>
        </a>
        <a
          href="/wallet/cosigners"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Cosigners
          </div>
        </a>
        <a
          href="/wallet/clients"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Clients
          </div>
        </a>
        <a
          href="/wallet/help"
        >
          <div
            class="p3_vx7tlw-o_O-pl5_qh1ng-o_O-flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-textTransformUppercase_4efw5a-o_O-fontWeight400_tt122m-o_O-wrap_s5wwyl"
          >
            Help
          </div>
        </a>
      </nav>
    </div>
    <div
      class="flex_jro6t0-o_O-flexGrow1_t0tx82-o_O-flexDirectionColumn_1jtvojh-o_O-mainContainer_1xmqyfw"
    >
      <div
        class="p5_1x929gn-o_O-pt3_1gv8f73-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-width100_zdxht7-o_O-container_1hoplq"
      >
        <div
          class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-width100_zdxht7-o_O-navbar_1u75k1v"
        >
          <h2
            class="textTransformUppercase_4efw5a"
          >
            PKS Wallet
          </h2>
          <div
            class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-height100_1yfus1e-o_O-pullRight_ls0e43"
          >
            <div
              class="mr4_kh8dwt-o_O-fontSize1_3kvou9-o_O-hidden_1u9fru1-o_O-notifications_e296pg"
            >
              <span
                class="lnr lnr-alarm"
              >
                
              </span>
              <div
                class="notificationCounterWrap_1k9zd85"
              >
                <div
                  class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-notificationCounter_1xicdc3"
                >
                  5
                </div>
              </div>
            </div>
            <div
              class="hidden_1u9fru1-o_O-separator_xrxmxy"
            >
              
            </div>
            <div
              class="ml4_1pgo4wm-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-alignItemsFlexEnd_1f2mvon"
            >
              <div
                class="textTransformUppercase_4efw5a-o_O-deviceName_1pe4kle"
              >
                Trezor Label
              </div>
              <div
                class="textTransformUppercase_4efw5a-o_O-deviceStatus_1wddarm"
              >
                connected
              </div>
            </div>
            <div
              class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-px2_6ebohg-o_O-ml1_1dn7v29-o_O-height100_1yfus1e-o_O-cursorPointer_13xlah4-o_O-dropdownButton_1w6iw80"
            >
              <span
                class="lnr lnr-chevron-down icon_ko4whp"
              >
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="pb3_1a70m23-o_O-content_14ddvtv"
        >
          <h1>
            Test header
          </h1>
        </div>
      </div>
      <div
        class="flexGrow1_t0tx82-o_O-flexShrink0_fz3zdn-o_O-p5_1x929gn-o_O-mainContainer_1xmqyfw"
      >
        <div>
          Test main content
        </div>
      </div>
      <footer
        class="px5_1g0n6pg-o_O-footerContainer_14ddvtv"
      >
        <div
          class="width100_zdxht7-o_O-py4_r2hutn-o_O-px1_17oj4c4-o_O-textTransformUppercase_4efw5a-o_O-footer_r1wilx"
        >
          2018 Â© PKS Wallet. All rights reserved.
        </div>
      </footer>
    </div>
  </div>
</div>
`;
